uxarray/grid/connectivity.py

```diff
@@ -230,3 +232,109 @@ def _build_face_edges_connectivity(grid):
             "start_index": INT_DTYPE(0),
             "long_name": "Maps every edge to the two nodes that it connects",
         })
+
+
+def _build_node_faces_connectivity(grid):
+    """Builds the `Grid.Mesh2_node_faces`: integer DataArray of size
+    (nMesh2_node, nMaxNumFacesPerNode) (optional) A DataArray of indices
+    indicating faces that are neighboring each node.
+
+    This function converts the face-node connectivity data into a sparse matrix, and then constructs the node-face
+    connectivity by iterating over each node in the mesh and retrieving the set of neighboring faces.
+
+    Raises
+    ------
+    RuntimeError
+        If the Mesh object does not contain a 'Mesh2_face_nodes' variable.
+    """
+    if "Mesh2_face_nodes" not in grid._ds:
+        raise RuntimeError(
+            "Currently, node_face_connectivity can only be built with Mesh2_face_nodes"
+        )
```

Comment:
The minimum set of variables always available are `Mesh2_node_x`, `Mesh2_node_y`, `Mesh2_face_nodes`, so this check shouldn't need to be necessary. 

Additionally, we should move these checks inside of the `Grid` property attribute in the future if they are needed.